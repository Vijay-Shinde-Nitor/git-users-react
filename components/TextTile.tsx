import { View, Text, StyleSheet } from "react-native"

export const TextTile = ({ title, data }: { title: string, data: string }) => {
    return (<View style={styles.container}>
        <Text style={styles.title}>{title}:</Text>
        <Text style={styles.data} numberOfLines={2}>{data}</Text>
    </View>);
};

const styles = StyleSheet.create({
    container: {
        flexDirection: "row",
        margin: 8
    },
    title: {
        fontWeight: "bold",
        fontSize: 18,
        paddingRight: 8,
    },
    data: {
        fontWeight: "normal",
        fontSize: 16,
        overflow: "hidden",
        paddingRight: 8        
    },
})