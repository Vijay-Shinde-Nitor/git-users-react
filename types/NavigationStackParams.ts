
export type RootStackParamList = {
    Login: undefined;
    UserDetails: { username: string };
    UserList: undefined;
};
