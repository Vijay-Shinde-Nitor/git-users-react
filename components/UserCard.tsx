import { useNavigation } from "@react-navigation/native";
import { NativeStackNavigationProp } from "@react-navigation/native-stack";
import React from "react";
import { StyleSheet, View, Text, ListRenderItemInfo, Image, TouchableHighlight } from "react-native";
import { ModelGitUser } from "../types/ModelGitUsers";
export type RootStackParamList = {
    UserDetails: { username: string };
};
export const UserCard = ({ item, clickable = true }: { item: ModelGitUser, clickable: boolean }) => {
    const navigation = useNavigation<NativeStackNavigationProp<RootStackParamList>>();
    return (
        <View style={styles.container}>
            <TouchableHighlight onPress={() => {
                if (clickable) {
                    navigation.navigate("UserDetails", { username: item.login });
                }
            }}
                underlayColor="#eee">
                <View style={{
                    flex: 1,
                    flexDirection: "row",
                    padding: 8,
                }}>
                    <View style={styles.col1}>
                        <Image
                            style={styles.img}
                            source={{ uri: item.avatar_url }}></Image>
                    </View>
                    <View style={styles.col2}>
                        <Text style={styles.title}>{item.login}</Text>

                    </View>
                </View>
            </TouchableHighlight>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        margin: 8,
        backgroundColor: "#fff",
        borderRadius: 8,
        // height: window.screen.availWidth*0.3
    }, col1: {
        flex: 0.2,
        padding: 8,
    },
    col2: {
        flex: 0.8,
        padding: 8,
    }, title: {
        fontWeight: "bold",
        fontSize: 20,
        paddingTop: 10,
        paddingBottom: 10,
        marginBottom: 5
    },
    img: {
        resizeMode: "cover",
        flex: 1,
        backgroundColor: "#eee",
    }
});