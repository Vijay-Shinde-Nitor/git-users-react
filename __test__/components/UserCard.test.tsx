import React, { Component } from "react";
import renderer from "react-test-renderer";
import { TextTile } from "../../components/TextTile";
import { UserCard } from "../../components/UserCard";
import { ModelGitUser } from "../../types/ModelGitUsers";

const model: ModelGitUser = {
    name: "Vijay SHinde",
    id: 1,
    login: "vijay",
    followers: 2,
    following: 3
}

const mockedNavigate = jest.fn();

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: mockedNavigate,
    }),
  };
});

test('renders correctly', () => {
    const tree = renderer.create(<UserCard item={model} clickable={false} />).toJSON();
    expect(tree).toMatchSnapshot();
});