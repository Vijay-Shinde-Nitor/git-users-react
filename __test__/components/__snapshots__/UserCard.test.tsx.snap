// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderRadius": 8,
      "flex": 1,
      "margin": 8,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "padding": 8,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 0.2,
            "padding": 8,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": undefined,
            }
          }
          style={
            Object {
              "backgroundColor": "#eee",
              "flex": 1,
              "resizeMode": "cover",
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 0.8,
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 20,
              "fontWeight": "bold",
              "marginBottom": 5,
              "paddingBottom": 10,
              "paddingTop": 10,
            }
          }
        >
          vijay
        </Text>
      </View>
    </View>
  </View>
</View>
`;
